src/elm/Page/Community/ObjectiveEditor.elm

```diff
@@ -411,6 +423,32 @@ update msg model loggedIn =
                     |> UR.addExt
                         (Just ClickedSaveObjective |> RequiredAuthentication)
 
+        ClickedArchiveObjetive ->
+            case model.objectiveId of
+                Just objectiveId ->
+                    model
+                        |> UR.init
+                        |> UR.addCmd
+                            (Api.Graphql.mutation loggedIn.shared
+                                (Mutation.completeObjective
+                                    { input = { objectiveId = objectiveId } }
+                                    empty
+                                )
+                                CompletedArchiveMutation
+                            )
+
+                Nothing ->
+                    model
+                        |> UR.init
+
+        CompletedArchiveMutation (Ok _) ->
+            model
```

Comment:
after the mutation is done, you can show a feedback based on the result, like we do [here](https://github.com/cambiatus/frontend/blob/cb6a6c9e2e4ef5a3a5fe15c3609eafa51e7aad74/src/elm/Page/Community/Editor.elm#L753-L790)